### ISSUE TITLE
[Bug]: UseFetcher breaks in JSDOM / Jest / testing-library environment

### ISSUE THREAD

**Issue by skwh** [remix-run/react-router#10955]
### What version of React Router are you using?

6.14.1

### Steps to Reproduce

Using `testing-library/react` v14, `jest` v29.6, `typescript` v5.1

```
import { render } from '@testing-library/react';
import { BrowserRouter, useFetcher } from 'react-router-dom';

const Component = () => {
  const fetcher = useFetcher();

  return (
    <fetcher.Form>
      Test
    </fetcher.Form>
  );
};

describe('Test Fetcher', () => {
  test('It renders as a child', () => {
    render(
      <BrowserRouter>
        <Component />
      </BrowserRouter>
    );
  });

  test('It renders with the wrapper option', () => {
    render(
      <Component />,
      { wrapper: BrowserRouter }
    );
  });
});

```

### Expected Behavior

UseFetcher hook detects the BrowserRouter wrapper and behaves as expected.

### Actual Behavior

Failed test + runtime error.

> useFetcher must be used within a data router.  See https://reactrouter.com/routers/picking-a-router.

Hook does not detect the browser router wrapper.

**Comment by akamfoad** [comment]
useFetcher is a hook for data router (available from 6.4+) [as mentioned in the docs](https://reactrouter.com/en/main/hooks/use-fetcher)

BrowserRouter doesn't support Data APIs, you can find out which routers to use from [Picking a Router page](https://reactrouter.com/en/main/routers/picking-a-router)

You might find [Migrating to RouterProvider](https://reactrouter.com/en/main/upgrading/v6-data) useful if you haven't migrated to Data APIs yet.

**Comment by skwh** [comment]
Thanks, I guess I misunderstood what was written on the Picking a Router page - I figured that `createBrowserRouter` and `BrowserRouter` were the same thing.
